cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/UserTaskState.java

```diff
@@ -69,14 +73,24 @@ private String getJSONString(CruiseControlParameters parameters) {
   }
 
   private void addJSONTask(List<Map<String, Object>> jsonUserTaskList,
-                           UserTaskManager.UserTaskInfo userTaskInfo) {
-    Map<String, Object> jsonObjectMap = new HashMap<>();
+                           UserTaskManager.UserTaskInfo userTaskInfo,
+                           boolean fetchCompletedTask) {
+    Map<String, Object> jsonObjectMap = new HashMap<>(fetchCompletedTask ? 6 : 5);
     String status = userTaskInfo.state().toString();
     jsonObjectMap.put(USER_TASK_ID, userTaskInfo.userTaskId().toString());
     jsonObjectMap.put(REQUEST_URL, userTaskInfo.requestWithParams());
     jsonObjectMap.put(CLIENT_ID, userTaskInfo.clientIdentity());
     jsonObjectMap.put(START_MS, Long.toString(userTaskInfo.startMs()));
     jsonObjectMap.put(STATUS, status);
+    // Populate original response of completed task if requested so.
+    if (fetchCompletedTask) {
+      try {
+        CruiseControlResponse response = userTaskInfo.futures().get(userTaskInfo.futures().size() - 1).get();
```

Comment:
The logic regarding getting the last future seems to have an implicit meaning -- is it possible to move this to a function with this expected name -- e.g. `completedTaskResponse()`?